osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -422,15 +420,15 @@ public void Explode(DrawableHitObject fruit)
                     fruit.Anchor = Anchor.TopLeft;
                     fruit.Position = caughtFruit.ToSpaceOfOtherDrawable(fruit.DrawPosition, ExplodingFruitTarget);
 
-                    caughtFruit.Remove(fruit);
+                    if (!caughtFruit.Remove(fruit))
+                        // we may have already been removed by a previous operation (due to the weird OnLoadComplete scheduling).
+                        // this avoids a crash on potentially attempting to Add a fruit to ExplodingFruitTarget twice.
```

Comment:
This really shouldn't be the case. Does this still happen if the issue I mentioned in my previous comment is resolved?

Reply:
Yeah, consider this scenario:

- Explode is called with a specific (non-last) hitobject, but it's not yet loaded
- Explode is called for all hitobjects on another object

First is "schedule", second isn't. Second runs and removes all hitobjects. First then runs and tries to re-remove the other hitobject

Reply:
Keep in mind: The schedule logic is shocking in the first place. This is just a band-aid fix until I figure a better way of handling this scenario to avoid the load-wait part.